### ISSUE TITLE
9 Patch editor in StyleBoxTexture does not work as expected

### ISSUE THREAD

**Issue by JoshuaSteward1193** [godotengine/godot#93419]
### Tested versions

Godot v4.2.2.stable 

### System information

macOS 14.5.0 - GLES3 (Compatibility) - Apple M1 Pro - Apple M1 Pro (10 Threads)

### Issue description

When trying to create a 9-Patch texture for a button, I have run into a difficulty in actually getting the texture to behave as a 9-Patch. When you click on the "Edit Region" button in the Sub-Region section of the style box editor, you are presented with your texture and four lines. (I am using a 32x32 pixelated texture). I did the natural thing, turned "Snap to Grid" on, drug all 4 lines 8 pixels inward (splitting the image into 9 slices) and find out only the center slice of the image is drawn onto the buttons.

I messed with every setting I could, until I finally came across this video: https://www.youtube.com/watch?v=zHXzlz4T9qE
Following his steps exactly, I was able to get the button to work the way one would expect a 9 Patch texture to work. I was curious what I did wrong, and it seems that where I went wrong was to drag the lines inside the image. From the presentation of the UI and what I expected those lines were for, I assumed that was how you delineated where you wanted your slices to happen. But the *actual* functionality of those lines seems to be to allow someone to upload a sprite sheet, and pick out which sprite they want to slice using those lines. And then, you must use the Texture Margin settings to actually draw your slices.

This may not be a bug and may be the intended behavior, but neither the UI nor the documentation seems to try to communicate how to exactly draw the slices. The documentation describes how the patch_margin fields work, but in the editor there are no fields for patch margin, only Texture Margin and Expand Margins.

Ideally, the GUI Edit Region would allow the user to drag their slice points to the desired location, because it (1) Provides instant feedback as to where the slices will occur, and (2) Eliminates guesswork with trying to put in the margins numerically in the editor. Also, and this may just be due to my prior experiences, but it is the "obvious" way to slice a texture.

### Steps to reproduce

1. Create button
2. Create new StyleBoxTexture
3. Add any texture to the StyleBox
4. Click on Texture -> Sub Region -> Edit Region
5. Drag the lines so that they cross into the texture
6. Observe that the button is only showing and stretching or tiling the center slice, and not all nine slices.

### Minimal reproduction project (MRP)

[StyleBoxErrorMRP.zip](https://github.com/user-attachments/files/15922560/StyleBoxErrorMRP.zip)

**Comment by KoBeWi** [comment]
Sub-Region Rect defines what part of the texture you want to use. Unless you want to use only part of the texture, it should be set to the default value of 0, which means whole texture.

The only properties relevant for 9 Patch are the Texture Margins.

**Comment by JoshuaSteward1193** [comment]
Right, that is what I suspected the intended behavior might be, but I do feel like the editor might could be a little clearer when it comes to communicating that to the user. There isn't even any mention of "9 Patch" anywhere in the StyleBoxTexture component, either in the documentation or the editor tooltips.

**Comment by liqites** [comment]
@JoshuaSteward1193 I might suffered the same problem, recently I download kenney UI which contains a 48 x 48 border texture, so I want to apply the border to a custom theme of Button. this is my solution finally solve the nine patch problem.
1. Create a new theme
2. Create a new Button Type
3. Create a new normal StyleBoxTexture
4. Edit StyleBoxTexture
5. In StyleBoxTexture inspector, drag texture into the texture property
6. set Texture Margins property to `left: 16px; top 16px; right: 16px; bottom: 16px`

why 16px ? since my texture size is 48 x 48 ,and it's a ninepatch texture which means should be 3 x 3 blcok. so 48 / 3 is 16px and it behaves as what I expected
![panel-000](https://github.com/godotengine/godot/assets/5523883/989a5efd-cb71-48d6-931e-8f4ecbaf7b7e)
.I will upload the border texture here.
